include/nbl/builtin/hlsl/bxdf/common.hlsl

```diff
@@ -568,10 +589,15 @@ struct SIsotropicMicrofacetCache
         return VdotH; 
     }
 
+    bool isTransmission() NBL_CONST_MEMBER_FUNC { return getVdotHLdotH() < scalar_type(0.0); }
```

Comment:
this might actually be wrong, unless we assert `sign(VdotH)==sign(NdotV)` and similar for `L` 

Problem is that BRDF GGX and friends can generate unmasked `H`, which is actually backfacing towards `L` (non VNDF variants can even generate `VdotH<0`).

The only question is then if that `L` can get used for anything later on (pdf=0 or more accurately mis_weight=0 should prevent that).

Reply:
btw this is also a problem with BSDF sampling, as fresnel can be high while reflection can be invalid, or low while refraction would be invalid too

P.S. we should set fresnel to 0 if the would be reflection is on the other hesmisphere, and 1 if refraction would be the same hesmiphere. I guess thats another reason to use posteriori Fresnel sampling.

Reply:
`NdotH>=0` is actually something `isValidMicrofacet` could assert